fineract-provider/src/main/java/org/apache/fineract/cob/common/InitialisationTasklet.java

```diff
@@ -41,11 +47,16 @@ public class InitialisationTasklet implements Tasklet {
 
     @Override
     public RepeatStatus execute(@NotNull StepContribution contribution, @NotNull ChunkContext chunkContext) throws Exception {
+        HashMap<BusinessDateType, LocalDate> businessDates = ThreadLocalContextUtil.getBusinessDates();
         AppUser user = userRepository.fetchSystemUser();
         UsernamePasswordAuthenticationToken auth = new UsernamePasswordAuthenticationToken(user, user.getPassword(),
                 new NullAuthoritiesMapper().mapAuthorities(user.getAuthorities()));
         SecurityContextHolder.getContext().setAuthentication(auth);
         ThreadLocalContextUtil.setActionContext(ActionContext.COB);
+        String businessDate = Objects.requireNonNull((String) chunkContext.getStepContext().getStepExecution().getJobExecution()
+                .getExecutionContext().get(LoanCOBConstant.BUSINESS_DATE_PARAMETER_NAME));
+        businessDates.put(BusinessDateType.COB_DATE, LocalDate.parse(businessDate, DateTimeFormatter.ISO_DATE));
```

Comment:
If we are about to support anyhow the SOB (Start of Business), we should update the other date (DEFAULT) as well, to the next day of the provided COB date. 
This way if one of the Business steps would like to use the "following day" instead of the COB date, they can easily do that by setting the ActionContext to DEFAULT.